comit/src/expiries/config.rs

```diff
@@ -0,0 +1,289 @@
+// gas price: current gas price can be learned from the Ethereum blockchain,
+//            for example from geth or infura.
+// modifier:  We could use a pre-configured modifier, as is widely done in the
+//            Ethereum ecosystem for slow, medium, fast.
+//
+// For Bitcoin:
+//
+// With a configured 'include within N blocks' we can use bitcoind via
+// bitcoin-cli, by calling:
+//
+//  `bitcoin-cli estimatesmartfee N_BLOCKS`
+//
+// CAVEAT: The fee estimator relies on an active mempool with good uptime to
+// watch fee activity on the network, that in turn implies that our calculations
+// herein are only as good as the fee estimator of the Bitcoin connector.
+
+const BITCOIN_BLOCK_TIME_SECS: usize = 600; // 10 minutes, average Bitcoin block time.
+const ETHEREUM_BLOCK_TIME_SECS: usize = 20; // Conservative Ethereum block time.
+const BITCOIN_MINE_WITHIN_N_BLOCKS: usize = 3; // Value arbitrarily picked.
+const ETHEREUM_MINE_WITHIN_N_BLOCKS: usize = 3;
+
+// TODO: Add support for lightning.
+#[derive(Clone, Copy, Debug)]
+pub enum Protocol {
+    Herc20Hbit,
+    HbitHerc20,
+}
+
+impl Protocol {
+    pub fn config(&self) -> Config {
+        // Since we are targeting Nectar and the App we can set these.
+        let alice = ActorType::Human;
+        let bob = ActorType::Bot;
+
+        match self {
+            Protocol::Herc20Hbit => Config {
+                alice,
+                bob,
+                alpha_required_confirmations: ethereum_required_confirmations(),
+                beta_required_confirmations: bitcoin_required_confirmations(),
+                alpha_block_time: ETHEREUM_BLOCK_TIME_SECS,
+                beta_block_time: BITCOIN_BLOCK_TIME_SECS,
+                alpha_mine_fund_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                beta_mine_fund_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                alpha_mine_redeem_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                beta_mine_redeem_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+            },
+            Protocol::HbitHerc20 => Config {
+                alice,
+                bob,
+                alpha_required_confirmations: bitcoin_required_confirmations(),
+                beta_required_confirmations: ethereum_required_confirmations(),
+                alpha_block_time: BITCOIN_BLOCK_TIME_SECS,
+                beta_block_time: ETHEREUM_BLOCK_TIME_SECS,
+                alpha_mine_fund_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                beta_mine_fund_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                alpha_mine_redeem_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                beta_mine_redeem_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+            },
+        }
+    }
+}
+
+/// Configuration values used during transition period calculations.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Config {
+    alice: ActorType,
+    bob: ActorType,
+    alpha_required_confirmations: usize,
```

Comment:
`usize` is an odd choice. Are you sure your not shooting yourself on the foot by using a platform dependent type?

Reply:
I thought `usize` was the default size to use for an unsigned integer in Rust. This value is always going to small (less than 2 bytes probably) so the integer size does not matter here.

Reply:
If you need to cast the `usize` variable to `u32` down the line due to another API that takes a `u32` then you risk a truncation that is only protected by the functional behaviour of the code ("less than 2 bytes probably") which is never good.
I don't think you do such cast but why not just using `u16` or `u32` instead if you think it's "less than 2 bytes probably"?

Also, as per https://doc.rust-lang.org/std/primitive.usize.html, usize is:

> The pointer-sized unsigned integer type.

> The size of this primitive is how many bytes it takes to reference any location in memory. For example, on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes.

This is not the "default size to use for an unsigned integer in Rust". It is the type to use for referencing location in memory.

Which means you should use `usize` if a variable is used to index memory location. For example, to track the location of an item in a `Vec` or other heaped container.

Reply:
Thanks, I consider myself schooled. Here is another quote by a Rust guy that backs up what you said. Thanks, this was a useful lesson.

> Use usize and isize when it’s related to memory size – the size of an object, or indexing a vector, for instance. It will be a 32-bit number on 32-bit platforms, as that’s the limit of memory they can address, and likewise for 64-bit.
>
> Use u32 and i32 when you just want numbers.

ref: https://users.rust-lang.org/t/i32-vs-isize-u32-vs-usize/22657